statsmodels/base/transform.py

```diff
@@ -0,0 +1,218 @@
+
+    def untransform_boxcox(self, x, lmbda, method='naive'):
+        """
+        Back-transforms the Box-Cox transformed data array, by means of the
+        indicated method. The provided argument lmbda should be the lambda
+        parameter that was used to initially transform the data.
+
+        Parameters
+        ----------
+        x : array_like
+            The transformed series.
+        lmbda : float
+            The lambda parameter that was used to transform the series.
+        method : {'naive', 'normal'}
+            Indicates the method to be used in the untransformation. Defaults
+            to 'naive', which reverses the transformation. 'normal' yields an
+            optimal back-transform, assuming the transform series is normally
+            distributed.
+
+        Returns
+        -------
+        y : array_like
+            The untransformed series.
+        """
+        method = method.lower()
+        x = np.asarray(x)
+
+        if method == 'naive':
+            if np.isclose(lmbda, 0.):
+                y = np.exp(x)
+            else:
+                y = np.power(lmbda * x + 1, 1. / lmbda)
+        elif method == 'normal':
+            y = x  # TODO
+        else:
+            raise ValueError("Method '{0}' not understood.".format(method))
+
+        return y
+
+    def _est_lambda(self, x, bounds=(-1, 2),
+                    R=2, method='guerrero', scale='sd'):
+        """
+        Computes an estimate for the lambda parameter in the Box-Cox
+        transformation using method.
+
+        Parameters
+        ----------
+        x : array_like
+            The untransformed data.
+        bounds : tuple
+            Numeric 2-tuple, that indicate the solution space for the lambda
+            parameter. Default (-1, 2).
+        R : int
+            The seasonality/grouping parameter. Default 2.
+        method : {'guerrero', 'loglik'}
+            The method by which to estimate lambda. Defaults to 'guerrero', but
+            the profile likelihood ('loglik') is also available.
+        scale : {'sd', 'mad'}
+            The dispersion measure to be used. 'sd' indicates the sample standard
+            deviation, but the more robust 'mad' is also available.
+
+        Returns
+        -------
+        lmbda : float
+            The lambda parameter.
+        """
+        method = method.lower()
+
+        if len(bounds) != 2:
+            raise ValueError("Bounds of length {0} not understood."
+                             .format(len(bounds)))
+        elif bounds[0] >= bounds[1]:
+            raise ValueError("Lower bound exceeds upper bound.")
+
+        if method == 'guerrero':
+            lmbda = self._guerrero_cv(x, R, bounds, scale=scale)
+        elif method == 'loglik':
+            lmbda = self._loglik(x, bounds)
+        else:
+            raise ValueError("Method '{0}' not understood.".format(method))
+
+        return lmbda
+
+    def _guerrero_cv(self, x, R, bounds, scale='sd'):
```

Comment:
This is a private function, in a quite specific (transforms) setting. Want me to change it, or leave it be as it is?